pkg/update/update.go

```diff
@@ -535,29 +528,23 @@ func updateDnsPod(httpClient *http.Client, domain, host, token, ip string) (newI
 		return "", err
 	}
 	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
-	resp, err = httpClient.Do(req)
+	_, content, err = network.DoHTTPRequest(httpClient, req)
 	if err != nil {
 		return "", err
 	}
-	respBody, err = ioutil.ReadAll(resp.Body)
-	resp.Body.Close()
-	if err != nil {
-		return "", err
-	}
-	type DDNSResp struct {
+	ddnsResp := new(struct {
 		Record struct {
 			ID    int64  `json:"id"`
 			Value string `json:"value"`
 			Name  string `json:"name"`
 		} `json:"record"`
-	}
-	ddnsResp := new(DDNSResp)
-	err = json.Unmarshal(respBody, ddnsResp)
+	})
+	err = json.Unmarshal(content, ddnsResp)
 	if err != nil {
 		return "", err
 	}
 	if ddnsResp.Record.Value != ip {
-		return "", errors.New("ip not set")
+		return "", fmt.Errorf("ip not set")
 	}
-	return ddnsResp.Record.Value, nil
+	return "", nil
```

Comment:
Please do not return an empty string, simply return the error. So change the function signature to return an error only and not a IP string and an error.